src/arch/arm/64/kernel/vspace.c

```diff
@@ -1354,6 +1387,15 @@ static exception_t decodeARMVSpaceRootInvocation(word_t invLabel, word_t length,
             return EXCEPTION_SYSCALL_ERROR;
         }
 
+#ifndef CONFIG_ARM_HYPERVISOR_SUPPORT
+        /* When in EL1, the mapping must be write-able for ARMVSpaceInvalidate_Data */
+        if (invLabel == ARMVSpaceInvalidate_Data && vmRightsFromPTE(pte) != VMReadWrite) {
```

Comment:
Strictly speaking it only faults if it's `VMReadOnly`. Only use case would be to explicitly give a task rights to invalidate a memory range without having direct access to the content, not sure if that's useful.